### ISSUE TITLE
[Accessibility]: Site admin sidebar: Focus lost when selecting item

### ISSUE THREAD

**Issue by umpox** [sourcegraph/sourcegraph#35866]
### Audit type

Keyboard navigation

### User journey audit issue

https://github.com/sourcegraph/sourcegraph/issues/34430

### Problem description

<img width="297" alt="image" src="https://user-images.githubusercontent.com/9516420/169831422-4b75c090-12aa-4a03-9c33-7066672b9f06.png">

1. Using only the <kbd>tab</kbd> and <kbd>enter</kbd> keys select an item in the sidebar.
2. Focus is lost, and users need to manually navigate back to the selected item.

### Expected behavior

Focus should stay on the currently selected item when clicked

### Assigning labels

- [X] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [X] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [ ] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [ ] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

This issue will be fixed by my team, I have assigned a relevant member to it, or I will do so in the near future.
